cli/mender-artifact/modify.go

```diff
@@ -41,61 +46,40 @@ func modifyArtifact(c *cli.Context) error {
 		return cli.NewExitError("File ["+c.Args().First()+"] does not exist.", 1)
 	}
 
-	candidateType, modifyCandidates, err :=
-		getCandidatesForModify(c.Args().First())
+	image, err := virtualImage.Open(comp, privateKey, c.Args().First())
 
 	if err != nil {
 		return cli.NewExitError("Error selecting images for modification: "+err.Error(), 1)
 	}
-	// strip the data and boot partitions
-	if candidateType == RootfsImageArtifact {
-		modifyCandidates = modifyCandidates[0:1]
-		for _, mc := range modifyCandidates {
-			defer os.Remove(mc.path)
-		}
-	} else if candidateType == RawSDImage {
-		if len(modifyCandidates) == 4 { // sdimg
-			defer func(partitions []partition) {
-				for _, part := range partitions {
-					os.Remove(part.path)
-				}
-			}(modifyCandidates)
-			modifyCandidates = modifyCandidates[1:3]
+	defer func() {
+		if err == nil {
+			err = image.Close()
```

Comment:
Question: This will not error in the case that one fails to close the image, as only cli exit errors does actually cause the application to return not zero to the shell. Is this intended?

Reply:
No, good catch!